### PR TITLE
feat(core): add email_template and liquid_template models

### PR DESCRIPTION
**Pull request by wrmk**
Part of #275.

- [x] I have reviewed my code in "Files changed" tab.
- [x] I have re-read the issue and this PR fully resolves it.

### CODE REVIEW

File: app/models/liquid_template.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    %w[sender_full_name sender_first_name full_name first_name position company].freeze
+
+  def self.extract_attributes_from(current_member:, candidate:, position:)
+    {
+      "sender_full_name" => current_member.name,
+      "sender_first_name" => current_member.name.split.first,
+      "full_name" => candidate.full_name,
+      "first_name" => candidate.full_name.split.first,
+      "position" => position.name,
+      "company" => current_member.tenant.name
```
Comment (Lyams) [inline app/models/liquid_template.rb:14]: I suggest to change the order to alphabetical or to as figma mockup

File: app/models/liquid_template.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    end
+
+    error_messages
+  end
+
+  def present_variables
+    Liquid::ParseTreeVisitor.for(@template.root)
+                            .add_callback_for(Liquid::VariableLookup) do |node|
+      [node.name, *node.lookups].join(".")
+    end.visit.flatten.uniq.compact
```
Comment (Lyams) [inline app/models/liquid_template.rb:54]: Nitpick: I suggest changing the style code here to one more commonly used in the project. Something like this
```suggestion
    Liquid::ParseTreeVisitor
      .for(@template.root)
      .add_callback_for(Liquid::VariableLookup) do |node|
      [node.name, *node.lookups].join(".")
    end.visit.flatten.uniq.compact
```

**Review by Lyams — APPROVED** [review]

File: db/migrate/20241129084054_add_email_templates.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class AddEmailTemplates < ActiveRecord::Migration[7.1]
+  def change
+    create_table :email_templates do |t|
```
Comment (delavaler) [inline db/migrate/20241129084054_add_email_templates.rb:5]: @wrmk , @Lyams  why this table doesn't have tenant_id?

Comment (wrmk) [reply db/migrate/20241129084054_add_email_templates.rb:5]: will add

Comment (wrmk) [reply db/migrate/20241129084054_add_email_templates.rb:5]: https://github.com/freeats/freeats/pull/284

**Pull request merged**